### DISCUSSION TITLE
Compiling NAPS2 with Visual Studio 2022

### DISCUSSION THREAD

**Discussion by m1bcodes** [cyanfish/naps2#55] [Development]
I have .NET 4.6.2 and Xamarin installed but when building the solution I get the following errors:

Severity	Code	Description	Project	File	Line	Suppression State

Error	MSB4019	The imported project "C:\Program Files\dotnet\sdk\7.0.101\Sdks\Microsoft.NET.Sdk\13.0.2032\targets\Xamarin.Shared.Sdk.MultiTarget.targets" was not found. Confirm that the expression in the Import declaration ";..\13.0.2032\targets\Xamarin.Shared.Sdk.MultiTarget.targets" is correct, and that the file exists on disk.	NAPS2.Sdk	C:\Program Files\dotnet\sdk\7.0.101\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets	1220	

Error	MSB4019	The imported project "C:\Program Files\dotnet\sdk\7.0.101\Sdks\Microsoft.NET.Sdk\13.0.2032\targets\Xamarin.Shared.Sdk.MultiTarget.targets" was not found. Confirm that the expression in the Import declaration ";..\13.0.2032\targets\Xamarin.Shared.Sdk.MultiTarget.targets" is correct, and that the file exists on disk.	NAPS2.Lib	C:\Program Files\dotnet\sdk\7.0.101\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets	1220	

Error		Project "C:\Program Files\dotnet\sdk\7.0.101\Sdks\Microsoft.NET.Sdk\13.0.2032\targets\Xamarin.Shared.Sdk.MultiTarget.targets" was not imported by "C:\Program Files\dotnet\sdk\7.0.101\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets" at (1220,3), due to the file not existing.	NAPS2.Images.Mac	C:\Program Files\dotnet\sdk\7.0.101\Sdks\Microsoft.NET.Sdk\13.0.2032\targets\Xamarin.Shared.Sdk.MultiTarget.targets	1220	

Error		Project "C:\Program Files\dotnet\sdk\7.0.101\Sdks\Microsoft.NET.Sdk\13.0.2032\targets\Xamarin.Shared.Sdk.MultiTarget.targets" was not imported by "C:\Program Files\dotnet\sdk\7.0.101\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets" at (1220,3), due to the file not existing.	NAPS2.Sdk	C:\Program Files\dotnet\sdk\7.0.101\Sdks\Microsoft.NET.Sdk\13.0.2032\targets\Xamarin.Shared.Sdk.MultiTarget.targets	1220	

Error		Project "C:\Program Files\dotnet\sdk\7.0.101\Sdks\Microsoft.NET.Sdk\13.0.2032\targets\Xamarin.Shared.Sdk.MultiTarget.targets" was not imported by "C:\Program Files\dotnet\sdk\7.0.101\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets" at (1220,3), due to the file not existing.	NAPS2.Lib	C:\Program Files\dotnet\sdk\7.0.101\Sdks\Microsoft.NET.Sdk\13.0.2032\targets\Xamarin.Shared.Sdk.MultiTarget.targets	1220	

Thanks a lot for any help,
Michael

**Comment by cyanfish** [comment]
Try running `dotnet workload install macos`. This seems to be related to https://github.com/xamarin/xamarin-macios/issues/16400.

**Comment by cyanfish** [comment]
I've also just consolidated development info into the Github [wiki](https://github.com/cyanfish/naps2/wiki), you may find that helpful.

**Comment by m1bcodes** [comment]
That worked!
Thanks a lot!
